Sources/ApodiniDelta/Schema/SchemaReference.swift

```diff
@@ -0,0 +1,31 @@
+//
+//  File.swift
+//  
+//
+//  Created by Eldi Cano on 21.03.21.
+//
+
+import Foundation
+
+/// A reference to a schema holding the schema name
+struct SchemaReference: Equatable, Hashable, Codable {
+    let schemaName: String
+}
+
+// MARK: - Convenience
+extension SchemaReference {
+    static func reference(_ named: String) -> SchemaReference {
+        SchemaReference(schemaName: named)
+    }
+
+    static func reference(_ primitiveType: PrimitiveType) -> SchemaReference {
+        .reference(primitiveType.description)
+    }
+
+    static var empty: SchemaReference {
+        .reference("")
+    }
```

Comment:
Good point, this will fall away once I have a centralised error handling logic. Currently instead of throwing I return `nil` and `empty` is a fallback if `ReflectionInfo` fails to obtain type info